core/main/src/bootstrap/extn/load_session_step.rs

```diff
@@ -32,6 +36,35 @@ impl Bootstep<BootstrapState> for LoadDistributorValuesStep {
     }
 
     async fn setup(&self, s: BootstrapState) -> RippleResponse {
+        // AccountSessionRequest::Get will not always be successful,
+        // on Error, it will set the ripple context activation status as not activated.
+        // On failure call activated state and send a telemetry event conveying any provisioning issue.
+        // Start a thread which will process the session token update event.
+        // Upon receving the event, if the ripple context is not set as active, set it to active.
+        //
+        //
+        let _resp = s
+            .platform_state
+            .get_client()
+            .send_extn_request(DeviceEventRequest {
+                event: DeviceEvent::DistributorSessionTokenChanged,
+                id: "internal".to_owned(),
```

Comment:
do we need this internal? By default all extensions talk back to Main

Reply:
Yes, this request is not coming from any app, it is initiated from Ripple itself for internal purposes. Hence chosen that name

Reply:
Hmm still trying to understand the need for that id given its not unique. What happens if we remove that field wouldn't SDK understand its internal feel it should be an Option and only added if the id is going to be unique

Reply:
Lets fix this by moving App id into fireboltAppEvent field